RiotSwiftUI/Modules/Authentication/Common/Service/MatrixSDK/AuthenticationService.swift

```diff
@@ -143,7 +143,7 @@ class AuthenticationService: NSObject {
     ///   - homeserverAddress: The homeserver to start the flow for, or `nil` to use the default.
     ///   If a provisioning link has been set, it will override the default homeserver when passing `nil`.
     func startFlow(_ flow: AuthenticationFlow, for homeserverAddress: String? = nil) async throws {
-        let address = homeserverAddress ?? provisioningLink?.homeserverUrl ?? BuildSettings.serverConfigDefaultHomeserverUrlString
+        let address = homeserverAddress ?? provisioningLink?.homeserverUrl ?? state.homeserver.addressFromUser ?? BuildSettings.serverConfigDefaultHomeserverUrlString
```

Comment:
Can we handle here the case where `state.homeserver.addressFromUser` is nil whereas `BuildSettings.forceHomeserverSelection` is true ? Perhaps this is not possible, but I want to be sure.
I would like to avoid using default url by mistake.
Can we force the Homeserver selection from here in that case?

Reply:
@giomfo This case is handled in the `AuthenticationCoordinator`, before calling this method, so there is no need to do it again here I think.